components/schema-migrator/migrations/director/20240127192653_constraint-validation.up.sql

```diff
@@ -0,0 +1,7 @@
+BEGIN;
+
+ALTER TABLE formation_template_constraint_references ADD CONSTRAINT unique_template_id_constraint_id UNIQUE (formation_template_id, formation_constraint_id);
```

Comment:
If there are multiple records for the same `formation_template_id <-> formation_constraint_id` the creation of this constraint will fail duplication error.

Have you checked if such "duplicated" records exist on all environments and overall what will be our strategy to ensure that the constraint creation won't fail?

Reply:
before the release such records should be removed manually. As the constraints are maintained by us there wont be any records created during the release